src/julia-syntax.scm

```diff
@@ -4885,21 +4896,18 @@ f(x) = yt(x)
     (compile e '() #t #t)
     (for-each (lambda (x)
                 (let ((point (car x))
-                      (hl    (cadr x))
-                      (src-tokens (caddr x))
+                      (src-handler-tokens (cadr x))
+                      (src-catch-tokens (caddr x))
                       (lab   (cadddr x)))
                   (let ((target-nesting (get label-nesting lab #f)))
                     (if (not target-nesting)
                         (error (string "label \"" lab "\" referenced but not defined")))
-                    (let ((target-level (car target-nesting)))
-                      (cond ((> target-level hl)
-                            (error (string "cannot goto label \"" lab "\" inside try/catch block")))
```

Comment:
Can we pass `lab` to the `pop-handler-list` function here, so that it can be preserved in the error message? I guess most functions shouldn't have many goto and labels, but still might be useful when the user is parsing a whole file and unsure which function it stuck on?